### PR TITLE
add overwrite flag for the output folder

### PR DESCRIPTION
**Pull request by tjungblu**

### CODE REVIEW

File: pkg/output/output.go [hunk 1]
```diff
@@ -48,29 +48,46 @@ func (f *fsWriter) Writer(parentPath, name string, permissions os.FileMode) (Clo
...
-// NewFSWriter returns an Outputter which writes the files and directories to a specified location.
-func NewFSWriter(path string) (Outputter, error) {
+func EnsureOutputPath(path string, deleteIfExists bool) error {
 	info, err := os.Stat(path)
 	if err != nil {
 		if !os.IsNotExist(err) {
-			return nil, err
+			return err
 		} else {
 			err := os.Mkdir(path, 0700)
```
Comment (arjunrn) [inline pkg/output/output.go:57]: You can return at this point and get rid of the else branch.

Comment (tjungblu) [reply pkg/output/output.go]: done!

File: pkg/output/output.go [hunk 2]
```diff
@@ -48,29 +48,42 @@ func (f *fsWriter) Writer(parentPath, name string, permissions os.FileMode) (Clo
...
+func EnsureOutputPath(path string, deleteIfExists bool) error {
 	info, err := os.Stat(path)
 	if err != nil {
-		if !os.IsNotExist(err) {
-			return nil, err
+		if os.IsNotExist(err) {
+			return os.Mkdir(path, 0700)
+		}
+		return err
+	} else {
```
Comment (arjunrn) [inline pkg/output/output.go:58]: You don't need to have the `else` branch anymore. Or rather whatever is in the else branch can be after the `if`. So the steps would be:
1. Stat the input path.
2. If error check if not exists and create if it does not. Either way return.
3. From stat info check if input is directory. if not return error. We don't want to accidentally delete something which is not an output directory.
4. check if directory is empty.
4.1 if not empty and delete flag is set delete
4.2 else return error
5. create directory
6. return

Comment (tjungblu) [reply pkg/output/output.go]: done

File: pkg/output/output.go [hunk 3]
```diff
@@ -55,31 +55,33 @@ func EnsureOutputPath(path string, deleteIfExists bool) error {
...
-				return fmt.Errorf("output destination must be a directory: '%s'", path)
-			}
-			entries, err := os.ReadDir(path)
-			if err != nil {
-				return fmt.Errorf("failed to get contents of output directory '%s': %w", path, err)
-			}
-			if len(entries) != 0 {
-				return fmt.Errorf("output directory %s is not empty", path)
-			}
+			return fmt.Errorf("output directory %s is not empty", path)
```
Comment (arjunrn) [inline pkg/output/output.go:76]: This should be after the `if deleteIfExists{}`

Comment (tjungblu) [reply pkg/output/output.go:76]: it is after the `deleteIfExists`. or do you mean to just leave the else statement out?

Comment (arjunrn) [reply pkg/output/output.go:76]: No this fine. I was reviewing in between updates.

**Pull request merged**